browser/src/model.js

```diff
@@ -0,0 +1,235 @@
+    this.resetLabels();
+    this.notifyInfoChange();
+  }
+
+  get channel() {
+    return this._channel;
+  }
+
+  set channel(value) {
+    // save current display settings before changing
+    this.adjuster.brightnessMap.set(this._channel, this.adjuster.brightness);
+    this.adjuster.contrastMap.set(this._channel, this.adjuster.contrast);
+    this.adjuster.invertMap.set(this._channel, this.adjuster.displayInvert);
+    // get brightness/contrast vals for new channel
+    this.adjuster.brightness = this.adjuster.brightnessMap.get(value);
+    this.adjuster.contrast = this.adjuster.contrastMap.get(value);
+    this.adjuster.displayInvert = this.adjuster.invertMap.get(value);
+
+    this._channel = value;
+    this.notifyInfoChange();
+  }
+
+  get highlight() {
+    return this._highlight;
+  }
+
+  set highlight(value) {
+    this._highlight = value;
+    this.notifyImageFormattingChange();
+  }
+
+  get rgb() {
+    return this._rgb;
+  }
+
+  set rgb(value) {
+    this._rgb = value;
+    this.adjuster.makeCascade();
+  }
+
+  get tracks() {
+    return this._tracks;
+  }
+
+  set tracks(value) {
+    this._tracks = value;
+    this.processMaxLabels();
+    this.notifyInfoChange();
+  }
+
+  get foreground() {
+    return this._foreground;
+  }
+
+  set foreground(value) {
+    this._foreground = value;
+    this.notifyInfoChange();
+    this.notifyImageFormattingChange();
+  }
+
+  get background() {
+    return this._background;
+  }
+
+  set background(value) {
+    this._background = value;
+    this.notifyInfoChange();
+    this.notifyImageFormattingChange();
+  }
+
+  get size() {
+    return this._size;
+  }
+
+  // set bounds on size of brush
+  set size(newSize) {
+    // don't need brush to take up whole frame
+    if (newSize > 0 && newSize < this.height / 2 && newSize < this.width / 2) {
+      this._size = newSize;
+      this.notifyImageChange();
+    }
+  }
+
+  resetLabels() {
+    this.foreground = this.maxLabelsMap.get(this.feature) + 1;
+    this.background = 0;
+  }
+
+  // TODO: use Observable interface instead of hard-coding the view as the only Observer
+  notifyImageChange() {
+    this.view.canvasView.render();
+  }
+
+  notifyImageFormattingChange() {
+    this.adjuster.preCompAdjust();
+  }
+
+  notifyInfoChange() {
+    this.view.infopaneView.render();
+  }
+
+  /**
+   * Updates maxLabelsMap to match the current tracks.
+   */
+  processMaxLabels() {
+    const tracks = this.tracks;
+    for (let i = 0; i < Object.keys(tracks).length; i++) {
+      const key = Object.keys(tracks)[i]; // the keys are strings
+      if (Object.keys(tracks[key]).length > 0) {
+        // use i as key in this map because it is an int, feature is also int
+        this.maxLabelsMap.set(i, Math.max(...Object.keys(tracks[key]).map(Number)));
+      } else {
+        // if no labels in feature, explicitly set max label to 0
+        this.maxLabelsMap.set(i, 0);
+      }
+    }
+  }
+
+  handlePayload(payload) {
+    if (payload.imgs) {
```

Comment:
Would it make sense to also use the `Object.prototype.hasOwnProperty` on `payload` checking for property `imgs`?

Reply:
On the Flask side, we set `imgs: false` when there are no images to send, so the payload always has this property.